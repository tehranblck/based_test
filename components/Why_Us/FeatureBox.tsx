import type { FeatureBoxProps } from "@/types";

export default function FeatureBox({ icon, title, description }: FeatureBoxProps) {
    return (
        <div className="group bg-card border border-border rounded-lg p-6 hover:shadow-xl hover:shadow-yellow-400/10 hover:-translate-y-1 hover:border-yellow-400/30 transition-all duration-300 ease-out cursor-pointer">
            <div className="mb-4 inline-flex items-center justify-center w-12 h-12 bg-yellow-400 rounded-lg group-hover:scale-110 group-hover:rotate-3 group-hover:bg-yellow-300 transition-all duration-300 ease-out">
                <div className="group-hover:scale-90 transition-transform duration-200 ease-out">
                    {icon}
                </div>
            </div>
            <h3 className="text-lg font-semibold text-foreground mb-3 group-hover:text-yellow-600 dark:group-hover:text-yellow-400 transition-colors duration-200">
                {title}
            </h3>
            <p className="text-sm text-muted-foreground leading-relaxed group-hover:text-foreground transition-colors duration-200">
                {description}
            </p>
        </div>
    );
}
