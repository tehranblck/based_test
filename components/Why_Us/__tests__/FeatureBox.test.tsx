import { render, screen } from "@testing-library/react";
import { Shield } from "lucide-react";
import FeatureBox from "../FeatureBox";

describe("FeatureBox", () => {
    const mockProps = {
        icon: <Shield data-testid="shield-icon" />,
        title: "Test Title",
        description: "Test description for the feature box component."
    };

    it("renders with all required props", () => {
        render(<FeatureBox {...mockProps} />);

        expect(screen.getByTestId("shield-icon")).toBeInTheDocument();
        expect(screen.getByText("Test Title")).toBeInTheDocument();
        expect(screen.getByText("Test description for the feature box component.")).toBeInTheDocument();
    });

    it("applies correct CSS classes for styling", () => {
        const { container } = render(<FeatureBox {...mockProps} />);
        const featureBox = container.firstChild as HTMLElement;

        expect(featureBox).toHaveClass("group", "bg-card", "border", "border-border", "rounded-lg", "p-6");
        expect(featureBox).toHaveClass("hover:shadow-xl", "hover:-translate-y-1", "transition-all", "duration-300", "cursor-pointer");
    });

    it("renders icon container with yellow background and hover animations", () => {
        const { container } = render(<FeatureBox {...mockProps} />);
        const iconContainer = container.querySelector(".bg-yellow-400");

        expect(iconContainer).toBeInTheDocument();
        expect(iconContainer).toHaveClass("inline-flex", "items-center", "justify-center", "w-12", "h-12", "rounded-lg");
        expect(iconContainer).toHaveClass("group-hover:scale-110", "group-hover:rotate-3", "transition-all");
    });

    it("renders title with correct typography and hover classes", () => {
        render(<FeatureBox {...mockProps} />);
        const title = screen.getByText("Test Title");

        expect(title).toHaveClass("text-lg", "font-semibold", "text-foreground", "mb-3");
        expect(title).toHaveClass("group-hover:text-yellow-600", "dark:group-hover:text-yellow-400", "transition-colors");
    });

    it("renders description with correct typography and hover classes", () => {
        render(<FeatureBox {...mockProps} />);
        const description = screen.getByText("Test description for the feature box component.");

        expect(description).toHaveClass("text-sm", "text-muted-foreground", "leading-relaxed");
        expect(description).toHaveClass("group-hover:text-foreground", "transition-colors");
    });

    it("handles empty title gracefully", () => {
        render(<FeatureBox {...mockProps} title="" />);

        expect(screen.getByTestId("shield-icon")).toBeInTheDocument();
        expect(screen.getByText("Test description for the feature box component.")).toBeInTheDocument();
    });

    it("handles empty description gracefully", () => {
        render(<FeatureBox {...mockProps} description="" />);

        expect(screen.getByTestId("shield-icon")).toBeInTheDocument();
        expect(screen.getByText("Test Title")).toBeInTheDocument();
    });

    it("renders different icons correctly", () => {
        const customIcon = <div data-testid="custom-icon">Custom</div>;
        render(<FeatureBox {...mockProps} icon={customIcon} />);

        expect(screen.getByTestId("custom-icon")).toBeInTheDocument();
        expect(screen.queryByTestId("shield-icon")).not.toBeInTheDocument();
    });

    it("has hover animation classes for interactive effects", () => {
        const { container } = render(<FeatureBox {...mockProps} />);
        const featureBox = container.firstChild as HTMLElement;
        const iconContainer = container.querySelector(".bg-yellow-400");
        const iconWrapper = iconContainer?.firstChild as HTMLElement;

        expect(featureBox).toHaveClass("group");
        expect(iconContainer).toHaveClass("group-hover:scale-110", "group-hover:rotate-3", "group-hover:bg-yellow-300");
        expect(iconWrapper).toHaveClass("group-hover:scale-90");
    });

    it("matches snapshot", () => {
        const { container } = render(<FeatureBox {...mockProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
