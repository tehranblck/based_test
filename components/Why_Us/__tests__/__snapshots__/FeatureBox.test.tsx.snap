// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FeatureBox matches snapshot 1`] = `
<div
  class="group bg-card border border-border rounded-lg p-6 hover:shadow-xl hover:shadow-yellow-400/10 hover:-translate-y-1 hover:border-yellow-400/30 transition-all duration-300 ease-out cursor-pointer"
>
  <div
    class="mb-4 inline-flex items-center justify-center w-12 h-12 bg-yellow-400 rounded-lg group-hover:scale-110 group-hover:rotate-3 group-hover:bg-yellow-300 transition-all duration-300 ease-out"
  >
    <div
      class="group-hover:scale-90 transition-transform duration-200 ease-out"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-shield"
        data-testid="shield-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 13c0 5-3.5 7.5-7.66 8.95a1 1 0 0 1-.67-.01C7.5 20.5 4 18 4 13V6a1 1 0 0 1 1-1c2 0 4.5-1.2 6.24-2.72a1.17 1.17 0 0 1 1.52 0C14.51 3.81 17 5 19 5a1 1 0 0 1 1 1z"
        />
      </svg>
    </div>
  </div>
  <h3
    class="text-lg font-semibold text-foreground mb-3 group-hover:text-yellow-600 dark:group-hover:text-yellow-400 transition-colors duration-200"
  >
    Test Title
  </h3>
  <p
    class="text-sm text-muted-foreground leading-relaxed group-hover:text-foreground transition-colors duration-200"
  >
    Test description for the feature box component.
  </p>
</div>
`;
